x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_logic.test.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+import { FlashMessagesLogic, IFlashMessage } from './flash_messages_logic';
+
+describe('FlashMessagesLogic', () => {
+  const DEFAULT_VALUES = {
+    messages: [],
+    queuedMessages: [],
+    historyListener: null,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    resetContext({});
+  });
+
+  it('has expected default values', () => {
+    FlashMessagesLogic.mount();
+    expect(FlashMessagesLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('setFlashMessages()', () => {
+    it('sets an array of messages', () => {
+      const messages = [
+        { type: 'success', message: 'Hello world!!' },
+        { type: 'error', message: 'Whoa nelly!', description: 'Uh oh' },
+        { type: 'info', message: 'Everything is fine, nothing is ruined' },
+      ] as IFlashMessage[];
```

Comment:
```suggestion
      const messages: IFlashMessage[] = [
        { type: 'success', message: 'Hello world!!' },
        { type: 'error', message: 'Whoa nelly!', description: 'Uh oh' },
        { type: 'info', message: 'Everything is fine, nothing is ruined' },
      ];
```

This is stricter and will have the compiler throw errors if for instance you had the wrong property name in one of the items in the array

Reply:
Oh super neat! Will def commit this change, although as a heads up I tend to be lazy about typing tests and go so far as `as any` especially with particularly annoying dependency mocks, so you might see me do this again in the future

Reply:
No worries, I just ran into a bug last week that I lost some time chasing my tail on because it wasn't caught by the the typescript compiler due to this (I added a prop to a kea logic but didn't update my call to create the logic to include the prop!), so its on the top of my mind.